### PR TITLE
chore: Add the cloud linter optional services

### PR DESCRIPTION
**Pull request by nand4011**
Add the Amazon services that cloud linter can optionally check, as well as calls it makes.

### CODE REVIEW

File: docs/cloud-linter/index.md [hunk 1]
```diff
@@ -56,6 +60,14 @@ DynamoDB:
 CloudWatch:
 - GetMetricData: 20 TPS
 
+S3:
+- ListBuckets
+- ListBucketMetricsConfigurations: 10 TPS
+
+API Gateway:
+- GetRestApis: 10 TPS
+- GetRestApi
```
Comment (nand4011) [inline docs/cloud-linter/index.md]: @pgautier404 I didn't see a limit on this call in the cloud linter code. Do you remember if that is done in a different way?

Comment (pgautier404) [reply docs/cloud-linter/index.md]: @nand4011 The methods for processing those resources are using the metrics and control plane limiters in use throughout. Let me know if that doesn't address the question and we can take a look at it.

**Review by cprice404 — COMMENTED** [review]
do we need to update the diagram at all? or are you thinking we can skip that because the diagram now reflects the default behavior?

**Comment by nand4011** [comment]
I updated the diagram.

File: docs/cloud-linter/index.md [hunk 2]
```diff
@@ -42,6 +42,10 @@ Cloud Linter is intended to interact with the following AWS services via their r
 - Amazon ElastiCache
 - Amazon CloudWatch
 
+It can optionally interact with the following AWS services:
+- Amazon S3, using the `-enable-s3` option
+- Amazon API Gateway, using the `-enable-api-gateway` option
```
Comment (pgautier404) [inline docs/cloud-linter/index.md]:
```suggestion
- Amazon S3, using the `--enable-s3` option
- Amazon API Gateway, using the `--enable-api-gateway` option
```

**Review by cprice404 — APPROVED** [review]

**Pull request merged**